external/fv3fit/tests/keras/test_halos.py

```diff
@@ -0,0 +1,159 @@
+from typing import Any, Dict, List
+from fv3fit.keras._models.shared import append_halos
+from fv3fit.keras._models.shared.halos import _append_halos_using_mpi
+import xarray as xr
+import numpy as np
+import pytest
+import fv3gfs.util
+
+
+def get_dataset(nx: int, ny: int, nz: int, n_tile: int) -> xr.Dataset:
+    nt = 5
+    ds = xr.Dataset(
+        data_vars={
+            "scalar": xr.DataArray(
+                data=np.ones([nt, n_tile, nx, ny]),
+                dims=["sample", "tile", "x", "y"],
+                attrs={"units": ""},
+            ),
+            "scalar_single_precision": xr.DataArray(
+                data=np.ones([nt, n_tile, nx, ny], dtype=np.float32),
+                dims=["sample", "tile", "x", "y"],
+                attrs={"units": ""},
+            ),
+            "column": xr.DataArray(
+                data=np.ones([nt, n_tile, nx, ny, nz]),
+                dims=["sample", "tile", "x", "y", "z"],
+                attrs={"units": ""},
+            ),
+        }
+    )
+    return ds
+
+
+@pytest.mark.parametrize(
+    "nx, ny, nz, n_tile, n_halo",
+    [
+        pytest.param(8, 8, 12, 6, 3, id="typical"),
+        pytest.param(8, 8, 12, 6, 1, id="one_halo"),
+        pytest.param(8, 8, 12, 6, 0, id="no_halo"),
+    ],
+)
+def test_append_halos_extends_dims(nx: int, ny: int, nz: int, n_tile: int, n_halo: int):
+    ds = get_dataset(nx=nx, ny=ny, nz=nz, n_tile=n_tile)
+    for name, da in ds.data_vars.items():
+        assert da.sizes["x"] == nx, name
```

Comment:
Later I check that the size of X is nx + 2 * n_halo. If the starting size isn't nx (which it should be unless the get_dataset helper has a bug), it will cause the test to fail. If there's a bug in get_dataset, I'd like it to show up here so we know the bug is in the test infrastructure and not the code.